writer.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 2022 by Posit Software, PBC
+package rsf
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type writer struct {
+	f io.Writer
+}
+
+func NewWriter(f io.Writer) Writer {
+	return &writer{f: f}
+}
+
+func (f *writer) WriteObject(v any) error {
+	var buf = &bytes.Buffer{}
+	var err error
+	err = f.writeObject(reflect.ValueOf(v), &tag{}, buf)
+	if err != nil {
+		return err
+	}
+
+	// Write size of full record
+	bs := make([]byte, 4)
+	recordSize := buf.Len() + 4
+	binary.LittleEndian.PutUint32(bs, uint32(recordSize))
+	_, err = f.f.Write(bs)
```

Comment:
Nit: some of the variable names could be more descriptive. Figuring out what the various `f`s and `r`s mean across `reader` and `writer` can be a little confusing at times.